src/types.rs

```diff
@@ -228,10 +234,16 @@ pub struct KakounePosition {
     pub column: u32, // in bytes, not chars!!!
 }
 
-#[derive(Clone, Debug)]
-pub struct HoverModal {
-    pub context: String,
-    pub do_after: String,
+#[derive(PartialEq)]
+pub enum HoverType {
```

Comment:
I wonder if there is a better name, like HoverUI, HoverTarget or HoverDestination.
Though these all sound awkward.